students/rachel_s/session02/grid_print.py

```diff
@@ -0,0 +1,101 @@
+def print_grid_arbitrary():
+    """Prints a grid of arbitrary size."""
+    print("+ - - - + - - - + - - - +")
+    for i in range(3):
+        print("|       |       |       |")
+    print("+ - - - + - - - + - - - +")
+    for i in range(3):
+        print("|       |       |       |")
+    print("+ - - - + - - - + - - - +")
+    for i in range(3):
+        print("|       |       |       |")
+    print("+ - - - + - - - + - - - +")
+
+
+def print_grid(n):
+    """Prints a grid of width and height n that is 2 cells high and wide."""
+    width = n // 2
+    # Print top border
+    print("+", end=" ")
+    for i in range(width):
+        print("-", end=" ")
+    print("+", end=" ")
```

Comment:
These 3 lines (lien 20-22) are repeated quite a few times in this function, so it would have been better if you created a helper function doing this and just call that function multiple times.

Reply:
Forgot to mention that there could be other such repeated code (possibly with slight differences that can be parameterized in functions) that could be refactored into subroutines (helper functions).